paymentsheet/src/main/java/com/stripe/android/link/ui/inline/InlineSignupViewState.kt

```diff
@@ -118,10 +116,27 @@ constructor(
                 fields = fields,
                 prefillEligibleFields = prefillEligibleFields,
                 isExpanded = isExpanded || allowsDefaultOptIn,
-                allowsDefaultOptIn = allowsDefaultOptIn,
+                allowsDefaultOptIn = allowsDefaultOptIn(config, signupMode),
+                linkSignUpOptInFeatureEnabled = linkSignupOptInFeatureEnabled(config, signupMode),
                 signUpState = signupState,
             )
         }
+
+        private fun allowsDefaultOptIn(
+            config: LinkConfiguration,
+            signupMode: LinkSignupMode
+        ): Boolean = config.allowDefaultOptIn &&
+            config.stripeIntent.countryCode == "US" &&
+            signupMode == LinkSignupMode.InsteadOfSaveForFutureUse
+
+        private fun linkSignupOptInFeatureEnabled(
+            config: LinkConfiguration,
+            signupMode: LinkSignupMode
+        ): Boolean = config.linkSignUpOptInFeatureEnabled &&
+            config.stripeIntent.countryCode == "US" &&
```

Comment:
Not needed for this feature, as it's included in `linkSignUpOptInFeatureEnabled`.